pkg/controller/core/workload_controller.go

```diff
@@ -310,7 +316,8 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 	// If the workload is admitted, updating the status here would set the Admitted condition to
 	// false before the workloads eviction.
 	if !workload.IsAdmitted(&wl) && workload.SyncAdmittedCondition(&wl, r.clock.Now()) {
-		if err := workload.ApplyAdmissionStatus(ctx, r.client, &wl, true, r.clock); err != nil {
+		err := workload.ApplyAdmissionStatus(ctx, r.client, &wl, true, r.clock)
+		if client.IgnoreNotFound(err) != nil {
 			return ctrl.Result{}, err
```

Comment:
```suggestion
		if err := workload.ApplyAdmissionStatus(ctx, r.client, &wl, true, r.clock); err != nil {
			return ctrl.Result{}, client.IgnoreNotFount(err)
```

This is also a new bug.